crates/apollo-mcp-server/src/operations/schema_walker.rs

```diff
@@ -0,0 +1,43 @@
+//! JSON Schema generation utilities
+//!
+//! The types in this module generate JSON schemas for GraphQL types by walking
+//! the types recursively.
+
+use apollo_compiler::{Schema as GraphQLSchema, ast::Type};
+use schemars::Schema;
+use serde_json::{Map, Value};
+
+use crate::custom_scalar_map::CustomScalarMap;
+
+pub(super) mod name;
+pub(super) mod r#type;
+
+/// Walk a top-level type and generate a JSON Schema
+pub fn walk(
```

Comment:
Does this function name clearly convey its purpose? 🤔 It might be confusing for someone new to the codebase. I think the previous name, `type_to_schema` was more intuitive.

Reply:
Hmm, yeah I can use that name again. I tried to convey that this process is recursive by using `walk`, but maybe that doesn't need to be conveyed in the name. I'll just add to the docs for this method that it might overflow the stack if the type is sufficiently nested.

As an aside, I mostly kept the existing implementation, but we should probably change this to not be recursive since we do not control the input here.